backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/RhosakEmailAggregator.java

```diff
@@ -1,10 +1,110 @@
+    private static final String SERVICE_DISRUPTION_EVENT_TYPE = "disruption";
+    private static final String CONTEXT = "context";
+    private static final String EVENTS = "events";
+    private static final String PAYLOAD = "events";
+
+    private static final String PAYLOAD_NAME = "name";
+    private static final String KAFKA_VERSION = "kafka_version";
+    private static final String SERVICE_DISRUPTION_IMPACT = "impacted_area";
+
+    // know service disruption impacts
+    private static final String PERFORMANCE = "performance";
+    private static final String LATENCY = "latency";
+    private static final String THROUGHPUT = "throughput";
+
+    public RhosakEmailAggregator() {
+        context.put(UPGRADES, new JsonObject());
+        context.put(SERVICE_DISRUPTIONS, new JsonObject());
+    }
+
     @Override
     void processEmailAggregation(EmailAggregation aggregation) {
-        // TODO after the payloads are determined, notably the scheduled upgrade payload
+        JsonObject aggregationPayload = aggregation.getPayload();
+        String eventType = aggregationPayload.getString(EVENT_TYPE);
+        JsonObject context = aggregationPayload.getJsonObject(CONTEXT);
+
+        if (SCHEDULED_UPGRADE_EVENT_TYPE.equals(eventType)) {
+            buildUpgradesPayload(aggregationPayload, context);
+        } else if (SERVICE_DISRUPTION_EVENT_TYPE.equals(eventType)) {
+            buildServiceDisruptionPayload(aggregationPayload, context);
+        }
+    }
+
+    private void buildServiceDisruptionPayload(JsonObject aggregationPayload, JsonObject context) {
+        String serviceDisruptionImpact = context.getString(SERVICE_DISRUPTION_IMPACT);
+        boolean currentDisruptionImpactPerformance = serviceDisruptionImpact.contains(PERFORMANCE);
+        boolean currentDisruptionImpactLatency = serviceDisruptionImpact.contains(LATENCY);
+        boolean currentDisruptionImpactTroughPut = serviceDisruptionImpact.contains(THROUGHPUT);
+
+        JsonObject disruptions = this.context.getJsonObject(SERVICE_DISRUPTIONS);
+        aggregationPayload.getJsonArray(EVENTS).stream().forEach(eventObject -> {
+            JsonObject event = (JsonObject) eventObject;
+            JsonObject payload = event.getJsonObject(PAYLOAD);
+            String name = payload.getString(PAYLOAD_NAME);
+            JsonObject serviceDisruption;
+            if (!disruptions.containsKey(name)) {
+                serviceDisruption = new JsonObject();
+                serviceDisruption.put(PAYLOAD_NAME, name);
+                serviceDisruption.put(SERVICE_DISRUPTION_IMPACT, serviceDisruptionImpact);
+                disruptions.put(name, serviceDisruption);
+            } else {
+                serviceDisruption = disruptions.getJsonObject(name);
+                String existingImpacts = serviceDisruption.getString(SERVICE_DISRUPTION_IMPACT, "");
+                boolean hasPerformanceImpact = existingImpacts.contains(PERFORMANCE);
```

Comment:
Should this be part of the dynamic payload?

Reply:
What do you mean? 

The `impacted_area` is part of the dynamic payload - see https://github.com/RedHatInsights/notifications-backend/blob/f6dcaf6f4d6a59266cbb828a3bc9bfb3992fed75/backend/src/main/resources/templates/Rhosak/serviceDisruptionBody.html#L21 for example. 

During aggregation we are merging all the impacts for a certain kafkas e.g 
The `existingImpacts` could be because there are two notifcation trigger for service disruption having the same kafka `x` one impacting `performance` and the trigger saying that `latency` was impacted. In the aggregated mailed we just want to say that for kafka `x`, service disruption impacted `performance, latency`

Reply:
thanks, make sense